stripe-eventbridge-lambda/README.md

```diff
@@ -0,0 +1,115 @@
+* [Create a Stripe account](https://stripe.com/register) if you do not already have one and log in.
+* [Set up the Stripe Amazon EventBridge SaaS integration](https://docs.stripe.com/event-destinations/eventbridge) if you have not already configured the integration.
+
+## Deployment Instructions
+
+1. Create a new directory, navigate to that directory in a terminal and clone the GitHub repository:
+    ``` 
+    git clone https://github.com/aws-samples/serverless-patterns
+    ```
+2. Change directory to the pattern directory:
+    ```
+    cd stripe-eventbridge-lambda
+    ```
+3. From the command line, use AWS CDK to deploy the AWS resources for the pattern as specified in the app.py file. A command-line argument is needed to deploy the CDK stack, "stripeEventBusName". This argument should be the name of the **SaaS event bus** associated with your Stripe [partner event source](https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-saas.html).
+    ```
+    cdk deploy --parameters stripeEventBusName=SAAS_EVENT_BUS_NAME_HERE
+    ```
+
+4. Note the outputs from the CDK deployment process. These contain the resource names and/or ARNs which are used for testing. This stack will output the name of the Lambda function deployed for testing to the CLI. See the example below.
+
+```
+Outputs:
+StripeIntegrationStack.StripeProcessFailedPaymentLambdaOutput = StripeIntegrationStack-StripeProcessFailedPaymentL-nIPhn7sPQx9e
+```
+
+## How it works
+
+This service interaction uses an existing Stripe SaaS integration in the customer's AWS account. If you do not have the Stripe SaaS integration set up in your AWS account, please set it up before deploying this pattern. View the [integration on the Stripe Docs](https://docs.stripe.com/event-destinations/eventbridge). See [Stripe's documentation](https://stripe.com/docs/api/events/types) for available events.
+
+This pattern demonstrates how to:
+1. Write EventBridge rules that match Stripe's event pattern
+2. Send events from Stripe's EventBridge SaaS integration to Amazon CloudWatch for logging and debugging
+3. Transform Stripe events using AWS Lambda
+4. Store failed payment details in a DynamoDB table
+
+See the below architecture diagram from the data flow of this pattern.
+
+![Architecture Diagram](./img/readme-arch-diagram.png)
+
+## Testing
+
+### Test the AWS Lambda Function
+
+The event.json file included in this pattern is a sample EventBridge event from Stripe. This event can be used to test the Lambda function and EventBridge rules deployed by this pattern.
+
+To test the Lambda function via the CLI, copy and paste the following command, replacing the variables in <> with your own values:
+```
+aws lambda invoke --function-name <StripeIntegrationStack-StripeProcessFailedPaymentL-nIPhn7sPQx9e> --payload file://event.json response.json
+```
+
+You should receive a response that looks like:
+```
+{
+    "StatusCode": 200,
+    "ExecutedVersion": "$LATEST"
+}
```

Comment:
I get an error message instead (account id redacted):

`{"statusCode": 500, "body": "An error occurred (AccessDeniedException) when calling the PutItem operation: User: arn:aws:sts::123456789012:assumed-role/StripeIntegrationStack-stripecdklambdarole246618B3-lLn1HaA8rFMz/StripeIntegrationStack-StripeProcessFailedPaymentL-BwLehI3Ed8BA is not authorized to perform: dynamodb:PutItem on resource: arn:aws:dynamodb:eu-west-1:123456789012:table/StripeIntegrationStack-StripeFailedPaymentsTableC8BBF5D3-SSP3IAPIWRPJ because no identity-based policy allows the dynamodb:PutItem action"}%`